draft-ietf-quic-applicability.md

```diff
@@ -566,42 +590,49 @@ to the server instance. The server can provide an IPv4 and an IPv6 address in a
-transport layer to avoid unnecessary load, as specified in {{Section 10.1.2 of
-QUIC}}. See {{resumption-v-keepalive}} for further guidance on keep-alives.
+# Connection Termination
+
+QUIC connections are terminated in one of three ways: implicit idle timeout,
+explicit immediate close, or explicit stateless reset.
+
+QUIC idle timeout is enabled via transport parameters. Client and server
+announce a timeout period and the effective value for the connection is the
+minimum of the two values. After the timeout period elapses, the connection is
+silently closed. An application therefore should be able to configure its own
+maximum value, as well as have access to the computed minimum value for this
+connection. An application may adjust the maximum idle timeout for new
+connections based on the number of open or expected connections, since shorter
+timeout values may free-up memory more quickly.
+
+As long as application data is exchanged, the idle timeout does not expire. If
+an application has no data to send but desires to keep the connection open for
+longer than the timeout period, it can send keep-alive messages. A QUIC
+implementation may provide an option to defer the time-out by sending keep-alive
+messages at the transport layer to avoid unnecessary load, as specified in
+{{Section 10.1.2 of QUIC}}. Alternatively, applications using QUIC could define
+their own mechanism, such as an application-layer ping, that achieves a similar
+result. See {{resumption-v-keepalive}} for further guidance on keep-alives.
+
+An immediate close is signalled by a CONNECTION_CLOSE frame (see
+{{error-handling}}). Immediate close causes all streams to become immediately
+closed. QUIC endpoints can manage the cumulative maximum number of streams they
+would allow to be opened using the MAX_STREAMS frames but there is no mechanism
+to reduce the value. An application that uses QUIC might commit to a number of
+openable streams but require the connection to be closed (for example, a
+scheduled maintenance period). Depending on how an application uses QUIC streams
+(see {{use-of-streams}}), abrupt closure of actively-used streams may be
+undesireable or detrimental. In contrast, waiting for an endpoint to exhaust the
+advertised limit may not suit application or operational needs. Applications
+using QUIC can use conservative stream limits and run to completion before
+enacting and immediate close. Alternatively, a graceful close mechanim can be
+used to commicate the intention to explicitly close the connection at some
+future point. QUIC does not provide any mechanism for graceful connection
+termination, applications using QUIC can define their own graceful termination
+process (see, for example, {{Section 5.2 of QUIC-HTTP}}).
+
+A stateless reset is an option of last resort for an endpoint that does not have
+access to connection state. It is not expected that application using QUIC need
+information or knowledge that a stateless reset was triggered.
```

Comment:
This is a little odd.  I would phrase the last sentence here more along the lines of "receiving a stateless reset is an indication of an unrecoverable error distinct from connection errors in that there is no application-layer information provided"